core/variables.js

```diff
@@ -259,72 +259,182 @@ Blockly.Variables.realizePotentialVar = function(varName, varType, potentialVarW
-                function() {
-                  promptAndCheckWithAlert(text);  // Recurse
-                });
-          } else {
-            var potentialVarMap = workspace.getPotentialVariableMap();
-            var variable;
-            // This check ensures that if a new variable is being created from a
-            // workspace that already has a variable of the same name and type as
-            // a potential variable, that potential variable gets turned into a
-            // real variable and thus there aren't duplicate options in the field_variable
-            // dropdown.
-            if (potentialVarMap && opt_type) {
-              variable = Blockly.Variables.realizePotentialVar(text, opt_type, workspace, false);
-            }
-            if (!variable) {
-              variable = workspace.createVariable(text, opt_type);
-            }
-
-            var flyout = workspace.isFlyout ? workspace : workspace.getFlyout();
-            var variableBlockId = variable.getId();
-            if (flyout.setCheckboxState) {
-              flyout.setCheckboxState(variableBlockId, true);
-            }
-
-            if (opt_callback) {
-              opt_callback(variableBlockId);
-            }
+        var validatedText = validate(text, workspace, opt_callback);
+        if (validatedText) {
+          // The name is valid according to the type, create the variable
+          var potentialVarMap = workspace.getPotentialVariableMap();
+          var variable;
+          // This check ensures that if a new variable is being created from a
+          // workspace that already has a variable of the same name and type as
+          // a potential variable, that potential variable gets turned into a
+          // real variable and thus there aren't duplicate options in the field_variable
+          // dropdown.
+          if (potentialVarMap && opt_type) {
+            variable = Blockly.Variables.realizePotentialVar(validatedText,
+                opt_type, workspace, false);
+          }
+          if (!variable) {
+            variable = workspace.createVariable(validatedText, opt_type);
+          }
+
+          var flyout = workspace.isFlyout ? workspace : workspace.getFlyout();
+          var variableBlockId = variable.getId();
+          if (flyout.setCheckboxState) {
+            flyout.setCheckboxState(variableBlockId, true);
+          }
+
+          if (opt_callback) {
+            opt_callback(variableBlockId);
           }
         } else {
           // User canceled prompt without a value.
           if (opt_callback) {
             opt_callback(null);
           }
         }
-      }, opt_type, modalTitle);
-  };
-  promptAndCheckWithAlert('');
+      }, modalTitle);
+};
+
+/**
+ * This function provides a common interface for variable name validation agnostic
+ * of type. This is so that functions like Blockly.Variables.createVariable and
+ * Blockly.Variables.renameVariable can call a single function (with a single type
```

Comment:
"type" is duplicated: on the end of this line and the start of the next line.

Reply:
done.